src/main/java/org/folio/circulation/domain/notice/TemplateRepository.java

```diff
@@ -0,0 +1,26 @@
+package org.folio.circulation.domain.notice;
+
+import java.util.concurrent.CompletableFuture;
+
+import org.folio.circulation.support.Clients;
+import org.folio.circulation.support.CollectionResourceClient;
+import org.folio.circulation.support.Result;
+import org.folio.circulation.support.http.client.Response;
+
+public class TemplateRepository {
+
+  private TemplateRepository(
+    CollectionResourceClient templateNoticeClient) {
+    this.templateNoticesClient = templateNoticeClient;
+  }
+
+  private final CollectionResourceClient templateNoticesClient;
```

Comment:
Personally, I would define fields above the constructor at the top of the class.